lib/scholar/metrics.ex

```diff
@@ -772,6 +790,48 @@ defmodule Scholar.Metrics do
     auc(fpr, tpr)
   end
 
+  @doc """
+  Compute the Brier score loss.
+
+  The smaller the Brier score loss, the better, hence the naming with "loss".
+  The Brier score measures the mean squared difference between the predicted
+  probability and the actual outcome. The Brier score always
+  takes on a value between zero and one, since this is the largest
+  possible difference between a predicted probability (which must be
+  between zero and one) and the actual outcome (which can take on values
+  of only 0 and 1). It can be decomposed as the sum of refinement loss and
+  calibration loss.
+
+  The Brier score is appropriate only for binary outcomes.
+
+  ## Options
+
+  #{NimbleOptions.docs(@brier_score_loss_schema)}
+
+  ## Examples
+
+      iex> y_true = Nx.tensor([0, 1, 1, 0])
+      iex> y_prob = Nx.tensor([0.1, 0.9, 0.8, 0.3])
+      iex> Scholar.Metrics.brier_score_loss(y_true, y_prob)
+      #Nx.Tensor<
+        f32
+        0.03750000149011612
+      >
+  """
+  deftransform brier_score_loss(y_true, y_prob, opts \\ []) do
+    if Nx.reduce_max(y_prob) > 1.0 == Nx.u8(1) or Nx.reduce_min(y_prob) < 0.0 == Nx.u8(1) do
```

Comment:
This is not going to work in a transform if the tensors are expressions. You can try having a `defn` that calls this function and I think it will always raise. Basically, inside a GPU, there is no way we can check for this. You should instead normalize the error to some value if out of range.

Reply:
Ok, so now we're clipping the values of predicted probabilities to [0,1] interval. I also added a note about it in the doc since it isn't intuitive behavior.